float32 offset
float32 vector_x
float32 vector_y
float32 angle
float32 target_x
float32 target_y